openedx/frontend-app-course-authoring
# feat: add duplicate file validation for asset upload

JIRA Ticket: [TNL-11087](https://2u-internal.atlassian.net/browse/TNL-11087)
> This functionality was recently added to `studio-frontend`. This needs functionality needs to be added to `frontend-app-course-authoring` to make sure all features reach parity before deprecating `studio-frontend`.

Testing

Cancel overwrite
1. Open the Files page
2. If the course does not have any assets add some
3. Click " + Add files"
4. Select files to upload and inlcude one that has been previously uploaded
5. Should see overwrite modal
6. Click "Cancel"
7. Check that all new files were uploaded
8. Find duplicate file
9. Open the info modal
10. Check the date and time added corresponds with the original upload


Overwrite file
3. Click " + Add files"
4. Select files to upload and inlcude one that has been previously uploaded
5. Should see overwrite modal
6. Click "Overwrite"
7. Check that all new files were uploaded
8. Find duplicate file
9. Open the info modal
10. Check the date and time added corresponds with the new upload
